src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -311,8 +320,8 @@ def acr_task_update(cmd,  # pylint: disable=too-many-locals
                             token_type=DEFAULT_TOKEN_TYPE
                         )
                     ),
-                    source_trigger_events=source_trigger_events,
-                    status=TriggerStatus.enabled.value if source_trigger_events else TriggerStatus.disabled.value,
+                    source_trigger_events=source_trigger_event,
```

Comment:
If this is a collection, you should definitely have an enum backing this, not a bunch of flags.  If you can only supply a single value currently, then allowing one value for that enum is okay. If, in the future, you can support multiple triggers, then that single enum can accept multiple values instead of a bunch of flags.

Reply:
@tjprescott do you want to consider a deprecation of the flag and introduce this? We initially through that breaking changes were discouraged.

Reply:
@sajayantony @jaysterp I would recommend introducing the enum in this PR and deprecating the flag.  You can deprecate the flag and hide it, but still allow it to continue to work (deferring the breaking change while still transitioning your customers).  

https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#deprecating-commands-and-arguments